openssl/0002-Support-MSYS2.patch

```diff
@@ -0,0 +1,273 @@
+diff --git a/apps/build.info b/apps/build.info
+index 751d8da..4c27432 100644
+--- a/apps/build.info
++++ b/apps/build.info
+@@ -22,7 +22,7 @@ IF[{- !$disabled{apps} -}]
+   INCLUDE[openssl]=.. ../include
+   DEPEND[openssl]=libapps.a ../libssl
+ 
+-IF[{- $config{target} =~ /^(?:Cygwin|mingw|VC-)/ -}]
++IF[{- $config{target} =~ /^(?:Cygwin|Msys|mingw|VC-)/ -}]
+   GENERATE[openssl.rc]=../util/mkrc.pl openssl
+   SOURCE[openssl]=openssl.rc
+ ENDIF
+diff --git a/build.info b/build.info
+index 3dda4e8..25e198a 100644
+--- a/build.info
++++ b/build.info
+@@ -71,7 +71,7 @@ ELSIF[{- $config{target} =~ /^vms/ -}]
+   SHARED_SOURCE[libssl]=libssl.opt
+ ENDIF
+ 
+-IF[{- $config{target} =~ /^(?:Cygwin|mingw|VC-)/ -}]
++IF[{- $config{target} =~ /^(?:Cygwin|Msys|mingw|VC-)/ -}]
+   GENERATE[libcrypto.rc]=util/mkrc.pl libcrypto
+   GENERATE[libssl.rc]=util/mkrc.pl libssl
+ 
+@@ -82,6 +82,9 @@ ENDIF
+ IF[{- $config{target} =~ /^Cygwin/ -}]
+  SHARED_NAME[libcrypto]=cygcrypto-{- $sover_filename -}
+  SHARED_NAME[libssl]=cygssl-{- $sover_filename -}
++ELSIF[{- $config{target} =~ /^Msys/ -}]
++ SHARED_NAME[libcrypto]=msys-crypto-{- $sover_filename -}
++ SHARED_NAME[libssl]=msys-ssl-{- $sover_filename -}
+ ELSIF[{- $config{target} =~ /^mingw/ -}]
+  SHARED_NAME[libcrypto]=libcrypto-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
+  SHARED_NAME[libssl]=libssl-{- $sover_filename -}{- $config{target} eq "mingw64" ? "-x64" : "" -}
+diff --git a/config b/config
+index b8adf34..02f3571 100755
+--- a/config
++++ b/config
+@@ -331,6 +331,10 @@ case "${SYSTEM}:${RELEASE}:${VERSION}:${MACHINE}" in
+ 	echo "${MACHINE}-pc-cygwin"; exit 0
+ 	;;
+ 
++    MSYS*)
++	echo "${MACHINE}-whatever-msys"; exit 0
++	;;
++
+     vxworks*)
+        echo "${MACHINE}-whatever-vxworks"; exit 0;
+        ;;
+@@ -815,6 +819,7 @@ case "$GUESSOS" in
+   # these are all covered by the catchall below
+   i[3456]86-*-cygwin) OUT="Cygwin-x86" ;;
+   *-*-cygwin) OUT="Cygwin-${MACHINE}" ;;
++  *-*-msys) OUT="Msys" ;;
+   x86-*-android|i?86-*-android) OUT="android-x86" ;;
+   armv[7-9]*-*-android)
+       OUT="android-armeabi"
+diff --git a/crypto/build.info b/crypto/build.info
+index 2c619c6..a196bcc 100644
+--- a/crypto/build.info
++++ b/crypto/build.info
+@@ -34,6 +34,6 @@ INCLUDE[armv4cpuid.o]=.
+ GENERATE[s390xcpuid.S]=s390xcpuid.pl $(PERLASM_SCHEME)
+ INCLUDE[s390xcpuid.o]=.
+ 
+-IF[{- $config{target} =~ /^(?:Cygwin|mingw|VC-)/ -}]
++IF[{- $config{target} =~ /^(?:Cygwin|Msys|mingw|VC-)/ -}]
+   SHARED_SOURCE[../libcrypto]=dllmain.c
+ ENDIF
+diff --git a/crypto/dllmain.c b/crypto/dllmain.c
+index 0838c55..f74d36c 100644
+--- a/crypto/dllmain.c
++++ b/crypto/dllmain.c
+@@ -10,8 +10,8 @@
+ #include "e_os.h"
+ #include "internal/cryptlib_int.h"
+ 
+-#if defined(_WIN32) || defined(__CYGWIN__)
+-# ifdef __CYGWIN__
++#if defined(_WIN32) || defined(__CYGWIN__) || defined(__MSYS__)
++# if defined(__CYGWIN__) || defined(__MSYS__)
```

Comment:
This is extra checks because ````__CYGWIN__```` is always defined for MSYS

Reply:
Good to know.